### PR TITLE
Options for file temperature for more files

### PR DESCRIPTION
**Pull request by pdillinger**
Summary: We have a request to use the cold tier as primary source of truth for the DB, and to best support such use cases and to complement the existing options controlling SST file temperatures, we add two new DB options:
* `metadata_write_temperature` for DB "small" files that don't contain much user data
* `wal_write_temperature` for WALs.

Test Plan: Unit test included, though it's hard to be sure we've covered all the places

### CODE REVIEW

File: db/db_impl/db_impl_open.cc [hunk 1]
```diff
@@ -319,6 +320,7 @@ Status DBImpl::NewDB(std::vector<std::string>* new_filenames) {
     }
     std::unique_ptr<FSWritableFile> file;
     FileOptions file_options = fs_->OptimizeForManifestWrite(file_options_);
+    file_options.temperature = immutable_db_options_.metadata_write_temperature;
```
Comment (pdillinger) [inline db/db_impl/db_impl_open.cc:323]: Perhaps I should not overwrite the result of `OptimizeForManifestWrite` if `metadata_write_temperature == kUnknown` (and similar below for `OptimizeForLogWrite`)

Comment (jowlyzhang) [reply db/db_impl/db_impl_open.cc:323]: Agreed, rest of the change is great! Thank you for adding this feature, maybe add a history entry too?

**Review by jowlyzhang — APPROVED** [review]
LGTM!

**Comment by cbi42** [comment]
How does read decide which temperature to use to open these files?

**Comment by pdillinger** [comment]
> How does read decide which temperature to use to open these files?

This would be committing pretty firmly to the model of temperatures being hints on the read side, and considering that relative to DB data size there are O(1) of these kinds of files (non-SST and non-blob), open-for-read time temperature hint is arguably not important.

We are already reasonably invested in the "hints only" model given the `*File::GetTemperature()` functions.

**Comment by cbi42** [comment]
> > How does read decide which temperature to use to open these files?
> 
> This would be committing pretty firmly to the model of temperatures being hints on the read side, and considering that relative to DB data size there are O(1) of these kinds of files (non-SST and non-blob), open-for-read time temperature hint is arguably not important.
> 
> We are already reasonably invested in the "hints only" model given the `*File::GetTemperature()` functions.

Ah, I wasn't aware that temperature is hints only, that makes sense.

**Pull request closed without merging**